packages/blockchain-utils-linking/src/cosmos.ts

```diff
@@ -0,0 +1,84 @@
+import { AccountID } from 'caip';
+import { AuthProvider } from './auth-provider';
+import { getConsentMessage, LinkProof } from './util';
+import { Tx, SignMeta } from '@tendermint/sig';
+import { hash } from '@stablelib/sha256';
+import * as uint8arrays from 'uint8arrays';
+
+const CHAIN_ID = "cosmos:cosmoshub-3";
+
+const stringEncode = (str: string): string => uint8arrays.toString(uint8arrays.fromString(str), 'base64pad');
+
+// return data in the cosmos unsigned transaction format
+function asTransaction(address: string, message: string): Tx {
+  return {
+    fee: {
+      amount: [{ amount: '0', denom: '' }],
+      gas: '0',
+    },
+    memo: message,
+    msg: [
+      {
+        type: 'cosmos-sdk/MsgSend',
+        value: {
+          from_address: address,
+          to_address: address,
+          amount: [{ amount: '0', denom: '0' }],
+        },
+      },
+    ],
+  };
+}
+
+// generate metadata for signing the transaction
+function getMetaData(): SignMeta {
+  return {
+    account_number: '1',
+    chain_id: 'cosmos',
+    sequence: '0',
+  };
+}
+
+export class CosmosAuthProvider implements AuthProvider {
+  readonly isAuthProvider = true;
+
+  constructor(
+    private readonly provider: any, 
+    private readonly address: string) {}
```

Comment:
Maybe you can pass the chainId (or chain reference, e.g. `cosmoshub-3`) here in the constructor?